hail/python/hail/matrixtable.py

```diff
@@ -3067,6 +3068,9 @@ def union_rows(*datasets: 'MatrixTable') -> 'MatrixTable':
                     raise ValueError(error_msg.format(
                         "col key types", 0, first.col_key.dtype, i+1, next.col_key.dtype
                     ))
+                if not hl.eval(next.col_key.collect(_localize=False) == col_keys):
```

Comment:
This will collect the keys of the first for each subsequent MT (_localize=False will inline the IR).

Could do something like:

`hl.rbind(first.col_key.collect(_localize=False), lambda first_keys: hl.case().when(hl.all(lambda other: other == first_keys, [mt.col_key.collect(_localize=False) for mt in datasets[1:]], True).or_error('error blah blah blah'))`